### ISSUE TITLE
`cargo install clippy` fails on latest nightly

### ISSUE THREAD

**Issue by Lytigas** [rust-lang-nursery/rust-clippy#2801]
I have zero background in this project, I just wanted to try it out for the first time. Anyway:

`rustup show`
```
Default host: x86_64-unknown-linux-gnu

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu (default)

installed targets for active toolchain
--------------------------------------

arm-unknown-linux-gnueabi
x86_64-unknown-linux-gnu

active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.28.0-nightly (990d8aa74 2018-05-25)
```

[`cargo install clippy --verbose`](https://gist.github.com/Lytigas/2b3299e63bbc3f7e2404ec7f4aed6d7a)

**Comment by Arnavion** [comment]
```
error[E0432]: unresolved import `rustc::mir::interpret::PrimVal`
   --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.204/src/consts.rs:418:33
    |
418 |     use rustc::mir::interpret::{PrimVal, ConstValue};
    |                                 ^^^^^^^ no `PrimVal` in `mir::interpret`
error[E0599]: no variant named `ByVal` found for type `rustc::mir::interpret::ConstValue<'_>` in the current scope
   --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.204/src/consts.rs:420:25
    |
420 |         ConstVal::Value(ConstValue::ByVal(PrimVal::Bytes(b))) => match result.ty.sty {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `rustc::mir::interpret::ConstValue<'_>`
error[E0599]: no variant named `ByValPair` found for type `rustc::mir::interpret::ConstValue<'_>` in the current scope
   --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.204/src/consts.rs:428:25
    |
428 |         ConstVal::Value(ConstValue::ByValPair(PrimVal::Ptr(ptr), PrimVal::Bytes(n))) => match result.ty.sty {
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `rustc::mir::interpret::ConstValue<'_>`
error: aborting due to 3 previous errors
```

**Comment by oli-obk** [comment]
On it

**Comment by oli-obk** [comment]
Published as 0.0.205
